### PR TITLE
layout: Support fractional sizes in pixel calculation

### PR DESCRIPTION
**Pull request by robinlinden**
The main things being that now we support fractional sizes, and warn on things we don't handle.

### CODE REVIEW

File: layout/layout.cpp [hunk 1]
```diff
@@ -99,10 +99,19 @@ int to_px(std::string_view property, int const font_size) {
         return res;
     }
 
-    if (property.ends_with("em")) {
+    auto const parsed_length = std::distance(property.data(), parse_result.ptr);
+    auto const suffix = property.substr(parsed_length);
```
Comment (mkiael) [inline layout/layout.cpp:103]: I think they call this `unit` in the CSS spec.

**Review by mkiael — APPROVED** [review]

File: layout/layout.cpp (hunk 1 above)
Comment (robinlinden) [reply layout/layout.cpp:103]: Good catch! Will update the name to match the spec. Also need to add a workaround for gcc 10 not supporting from_chars with floats. :(

Comment (robinlinden) [reply layout/layout.cpp]: Right, renamed it and added a very not spec-compliant version of from_chars for gcc 10. Can you please have a look at the last commit again?

Comment (robinlinden) [reply layout/layout.cpp]: Also added a few tests to make sure we run a few more of the failure paths in the layout system.

Comment (mkiael) [reply layout/layout.cpp]: Looks good!

**Pull request merged**